paimon-core/src/main/java/org/apache/paimon/operation/AbstractFileStoreScan.java

```diff
@@ -316,19 +320,27 @@ private boolean filterManifestFileMeta(ManifestFileMeta manifest) {
     }
 
     /** Note: Keep this thread-safe. */
-    private boolean filterByBucket(ManifestEntry entry) {
+    private boolean filterByBucket(AbstractManifestEntry entry) {
         return (specifiedBucket == null || entry.bucket() == specifiedBucket);
     }
 
     /** Note: Keep this thread-safe. */
-    private boolean filterByBucketSelector(ManifestEntry entry) {
+    private boolean filterByBucketSelector(AbstractManifestEntry entry) {
         return (bucketSelector == null
                 || bucketSelector.select(entry.bucket(), entry.totalBuckets()));
     }
 
     /** Note: Keep this thread-safe. */
-    private boolean filterByLevel(ManifestEntry entry) {
-        return (levelFilter == null || levelFilter.test(entry.file().level()));
+    private boolean filterByLevel(AbstractManifestEntry entry) {
+        return (levelFilter == null || levelFilter.test(entry.level()));
+    }
+
+    /** Note: Keep this thread-safe. */
+    private boolean filterByStats(AbstractManifestEntry entry) {
+        if (entry instanceof ManifestEntry) {
+            return filterByStats((ManifestEntry) entry);
+        }
+        return true;
```

Comment:
Add comments here: `filterByStats is an action that is completed as much as possible and does not have an impact if it is not done.`

Reply:
done